### ISSUE TITLE
Request::getUri() returns parameters in wrong order

### ISSUE THREAD

**Issue by juhasev** [laravel/framework#30844]
- Laravel Version: 6.5
- PHP Version: 7.3
- Database Driver & Version:

### Description:
Request::getUri() returns parameters in wrong order. This maybe because $request->getQueryString() also returns URL parameters in wrong order. 

Original URL:

```
https://test.app/p/exit?id=00000000-0000-0000-0000-000000000000&s=c&hash=952f21e5f3a5f9338191b8f0028af89307f8bfb382bc560d8914dea83da63f3385bf4ce3c15f892dd07b5cd1fa0658fe0b51af8d22e2f59581b664
```

getUri() returns parameters in wrong order:

```
https://test.app/p/exit?hash=952f21e5f3a5f9338191b8f0028af89307f8bfb382bc560d8914dea83da63f3385bf4ce3c15f892dd07b5cd1fa0658fe0b51af8d22e2f59581b664&id=00000000-0000-0000-0000-000000000000&s=c
```

getQueryString() also return parameters in wrong order:
```
hash=952f21e5f3a5f9338191b8f0028af89307f8bfb382bc560d8914dea83da63f3385bf4ce3c15f892dd07b5cd1fa0658fe0b51af8d22e2f59581b664&id=00000000-0000-0000-0000-000000000000&s=c
```

### Steps To Reproduce:
Pass some random params to any route and view output of $request->getUri()

**Comment by juhasev** [comment]
The only work around is to use the following:

$paramsInRightOrder = $request->getSchemeAndHttpHost() . $request->getRequestUri();

**Comment by mfn** [comment]
What is you use-case?

**Comment by juhasev** [comment]
Basically we need to rely on partner generated URLs that are secured with hash to prevent tampering. The hash value is computed from all the other parameters in correct order and appended at the end of the URL.

**Comment by mfn** [comment]
If you need this, you need to take it up with Symfony, where `getQueryString` is actually defined => `\Symfony\Component\HttpFoundation\Request::getQueryString`

I doubt however something will change, see the method `\Symfony\Component\HttpFoundation\Request::normalizeQueryString` from therein:
```php
        parse_str($qs, $qs);
        ksort($qs);
```
They you have your answer already.

TL;DR: if you want to rely on URL as a cash key, normalize it first. You could even call `normalizeQueryString` directly, it's public

**Comment by driesvints** [comment]
I'm afraid @mfn is right and that this either needs to be taken up with Symfony. In general the order of query string parameters shouldn't matter. 

You can btw make use of signed urls for your use case: https://laravel.com/docs/6.x/urls#signed-urls
